src/infra/dex/okx/dto.rs

```diff
@@ -0,0 +1,322 @@
+//! DTOs for the OKX swap API. Full documentation for the API can be found
+//! [here](https://www.okx.com/en-au/web3/build/docs/waas/dex-swap).
+
+use {
+    crate::{
+        domain::{dex, order},
+        util::serialize,
+    },
+    bigdecimal::BigDecimal,
+    ethereum_types::{H160, U256},
+    serde::{Deserialize, Serialize},
+    serde_with::serde_as,
+};
+
+/// A OKX API swap request parameters.
+///
+/// See [API](https://www.okx.com/en-au/web3/build/docs/waas/dex-swap)
+/// documentation for more detailed information on each parameter.
+#[serde_as]
+#[derive(Clone, Default, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SwapRequest {
+    /// Chain ID
+    #[serde_as(as = "serde_with::DisplayFromStr")]
+    pub chain_id: u64,
+
+    /// Input amount of a token to be sold set in minimal divisible units
+    #[serde_as(as = "serialize::U256")]
+    pub amount: U256,
+
+    /// Contract address of a token to be send
+    pub from_token_address: H160,
+
+    /// Contract address of a token to be received
+    pub to_token_address: H160,
+
+    /// Limit of price slippage you are willing to accept
+    pub slippage: Slippage,
+
+    /// User's wallet address
+    pub user_wallet_address: H160,
+
+    /// The fromToken address that receives the commission.
+    /// Only for SOL or SPL-Token commissions.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub referrer_address: Option<H160>,
+
+    /// Recipient address of a purchased token if not set,
+    /// user_wallet_address will receive a purchased token.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub swap_receiver_address: Option<H160>,
+
+    /// The percentage of from_token_address will be sent to the referrer's
```

Comment:
Probably makes sense to remove fields which don't get used in practice to keep the code lean and the surface area for breakages minimal.
Also applies to response data we don't need.